lib/commons/text/unicode.js

```diff
@@ -93,11 +93,12 @@ function getUnicodeNonBmpRegExp() {
 	 * '\u2600-\u26FF'  Misc Symbols
 	 * '\u2700-\u27BF'  Dingbats
 	 * '\uE000-\uF8FF'  Private Use
+	 * '\u{F0000}-\u{FFFFD}' Supplementary Private Use Area A
 	 *
 	 * Note: plane '\u2000-\u206F' used for General punctuation is excluded as it is handled in -> getPunctuationRegExp
 	 */
 
-	return /[\u1D00-\u1D7F\u1D80-\u1DBF\u1DC0-\u1DFF\u20A0-\u20CF\u20D0-\u20FF\u2100-\u214F\u2150-\u218F\u2190-\u21FF\u2200-\u22FF\u2300-\u23FF\u2400-\u243F\u2440-\u245F\u2460-\u24FF\u2500-\u257F\u2580-\u259F\u25A0-\u25FF\u2600-\u26FF\u2700-\u27BF\uE000-\uF8FF]/g;
+	return /[\u1D00-\u1D7F\u1D80-\u1DBF\u1DC0-\u1DFF\u20A0-\u20CF\u20D0-\u20FF\u2100-\u214F\u2150-\u218F\u2190-\u21FF\u2200-\u22FF\u2300-\u23FF\u2400-\u243F\u2440-\u245F\u2460-\u24FF\u2500-\u257F\u2580-\u259F\u25A0-\u25FF\u2600-\u26FF\u2700-\u27BF\uE000-\uF8FF\u{F0000}-\u{FFFFD}]/gu;
```

Comment:
Note the `u` flag added to the regex. This enables using unicode escape sequences in the pattern.

One side-effect of this is that the build step actually compiles the regex to this in the final output:

```js
function getUnicodeNonBmpRegExp() {
  return /(?:[\u1D00-\u1DFF\u20A0-\u27BF\uE000-\uF8FF]|[\uDB80-\uDBBE][\uDC00-\uDFFF]|\uDBBF[\uDC00-\uDFFD])/g;
}
```

The `u` flag is an ES6 feature, and I'm assuming that [preset-env is compiling to a regex that will work in older ES versions](https://babeljs.io/docs/en/babel-plugin-transform-unicode-regex).

Reply:
I think we should add this range to the `getSupplementaryPrivateUseRegExp` function. I also think the `text.hasUnicode` should check this range for the `nonBMP` flag (currently it's missing and that's probably a bug since `removeUnicode` with `nonBMP` uses it).

Reply:
Makes sense 👍